src/keywords.rs

```diff
@@ -1,170 +1,193 @@
-    )
-);
+fn keyword_o_to_s(i: &[u8]) -> IResult<&[u8], &[u8]> {
+    alt((
+        terminated(tag_no_case("ON"), keyword_follow_char),
+        terminated(tag_no_case("OR"), keyword_follow_char),
+        terminated(tag_no_case("OUTER"), keyword_follow_char),
+        terminated(tag_no_case("PLAN"), keyword_follow_char),
+        terminated(tag_no_case("PRAGMA"), keyword_follow_char),
+        terminated(tag_no_case("PRIMARY"), keyword_follow_char),
+        terminated(tag_no_case("QUERY"), keyword_follow_char),
+        terminated(tag_no_case("RAISE"), keyword_follow_char),
+        terminated(tag_no_case("RECURSIVE"), keyword_follow_char),
+        terminated(tag_no_case("REFERENCES"), keyword_follow_char),
+        terminated(tag_no_case("REGEXP"), keyword_follow_char),
+        terminated(tag_no_case("REINDEX"), keyword_follow_char),
+        terminated(tag_no_case("RELEASE"), keyword_follow_char),
+        terminated(tag_no_case("RENAME"), keyword_follow_char),
+        terminated(tag_no_case("REPLACE"), keyword_follow_char),
+        terminated(tag_no_case("RESTRICT"), keyword_follow_char),
+        terminated(tag_no_case("RIGHT"), keyword_follow_char),
+        terminated(tag_no_case("ROLLBACK"), keyword_follow_char),
+        terminated(tag_no_case("ROW"), keyword_follow_char),
+        terminated(tag_no_case("SAVEPOINT"), keyword_follow_char),
+        terminated(tag_no_case("SELECT"), keyword_follow_char),
+    ))(i)
+}
 
-named!(keyword_t_to_z<CompleteByteSlice, CompleteByteSlice>,
-    alt!(
-          terminated!(tag_no_case!("TABLE"), keyword_follow_char)
-        | terminated!(tag_no_case!("TEMP"), keyword_follow_char)
-        | terminated!(tag_no_case!("TEMPORARY"), keyword_follow_char)
-        | terminated!(tag_no_case!("THEN"), keyword_follow_char)
-        | terminated!(tag_no_case!("TO"), keyword_follow_char)
-        | terminated!(tag_no_case!("TRANSACTION"), keyword_follow_char)
-        | terminated!(tag_no_case!("TRIGGER"), keyword_follow_char)
-        | terminated!(tag_no_case!("UNION"), keyword_follow_char)
-        | terminated!(tag_no_case!("UNIQUE"), keyword_follow_char)
-        | terminated!(tag_no_case!("UPDATE"), keyword_follow_char)
-        | terminated!(tag_no_case!("USING"), keyword_follow_char)
-        | terminated!(tag_no_case!("VACUUM"), keyword_follow_char)
-        | terminated!(tag_no_case!("VALUES"), keyword_follow_char)
-        | terminated!(tag_no_case!("VIEW"), keyword_follow_char)
-        | terminated!(tag_no_case!("VIRTUAL"), keyword_follow_char)
-        | terminated!(tag_no_case!("WHEN"), keyword_follow_char)
-        | terminated!(tag_no_case!("WHERE"), keyword_follow_char)
-        | terminated!(tag_no_case!("WITH"), keyword_follow_char)
-        | terminated!(tag_no_case!("WITHOUT"), keyword_follow_char)
-    )
-);
+fn keyword_s_to_z(i: &[u8]) -> IResult<&[u8], &[u8]> {
+    alt((
+        terminated(tag_no_case("SET"), keyword_follow_char),
```

Comment:
"SET" could move back with the other 's' keywords.

Reply:
I actually don't think it can because the `alt` combinator can only support like 20 parsers at a time. (I think that might be because of const-generics not being fully complete??)

Reply:
You're right, I forgot about tuple sizing issues.  The Alt trait is implemented for tuples sizes up to 21.  

A comment explaining that SET isn't with the other s* elements because we hit a max tuple size for alt could help later devs avoid confusion.  

from nom::branch::mod
``` rust
/// helper trait for the [alt()] combinator
///
/// this trait is implemented for tuples of up to 21 elements
pub trait Alt<I, O, E> {
  /// tests each parser in the tuple and returns the result of the first one that succeeds
  fn choice(&self, input: I) -> IResult<I, O, E>;
}
```